### DISCUSSION TITLE
Why Headless UI has so little components compare to something like MUI

### DISCUSSION THREAD

**Discussion by kasir-barati** [tailwindlabs/headlessui#3140] [General]
Hi,

Thanks for reading and sharing your thought on the matter. I decided to use this lib for my react project and then I realized that I am still in need of other components such as `Input`, `Search`, etc. Is it intentional and if yes please tell me why exactly?

So here are the available component in MUI: https://mui.com/material-ui/all-components/

Thanks in advance :pray:.

**Comment by roomman** [comment]
@kasir-barati I think of headlessui as a the fundamental building blocks you would need to make a component library like Material UI. 

The unreleased version of headlessui (see the main branch) has some additional primitives, including an Input. That said, it sounds to me like you are looking for a component library like [Catalyst](https://tailwindui.com/templates/catalyst), [shadcn](https://ui.shadcn.com), or [DaisyUI](https://daisyui.com), which takes building blocks, like those offered by headlessui, to the next step? 

Catalyst is a great option because it implements headlessui and is made by Tailwind Labs. That said, it's still a development preview so if you're looking for a production ready solution you might need to look elsewhere. I hope that's helpful. 👍🏼

**Reply by kasir-barati** [reply]
So basically what does it mean is that I would be better of with MUI for the time being, specially considering my background (coming from backend world). But thanks for sharing you thought. But I did not know we have something like Catalyst. Good to know. Will probably give it a go sometime in the future.
